tests/TestCase/ORM/Query/SelectQueryTest.php

```diff
@@ -1432,6 +1433,24 @@ public function testFormatResultsBelongsToMany(): void
         $this->assertInstanceOf(DateTime::class, $first->tags[0]->created);
     }
 
+    public function testBelongsToManyWithPreservedKeys(): void
+    {
+        $table = $this->getTableLocator()->get('Articles');
+        $this->getTableLocator()->get('Tags', ['className' => TagsTable::class]);
+        $table->belongsToMany('Tags');
+
+        $first = $table->find()
+            ->where(['Articles.id' => 1])
+            ->contain([
+                'Tags' => ['finder' => 'slugged'],
```

Comment:
Is preservedKeys set to true by default?

Reply:
No, it's set in the finder, check `TagsTable::findSlugged()`.

Reply:
Ok. I saw the Articles slugged below and thought it odd we have 2 different slugged finders with different defaults. It's just tests, but was a little confusing to have to know.

Reply:
I purposely added 2 tests to show that one could set the option in the finder method itself or set it during a find() call.

I went with a finder option as it was the most convenient, another alternative would be to add an option/property for the association class and allow setting it as part of the association config.